include/deal.II/base/hdf5.h

```diff
@@ -326,6 +326,16 @@ DEAL_II_NAMESPACE_OPEN
  * thread-safe HDF5 version serializes the API but does not provide any level of
  * concurrency. To achieve high parallel performance with HDF5, we advice to use
  * HDF5 with MPI.
+ *
+ * # Exceptions and parallel HDF5
+ * In certain corner cases, exceptions combined with parallel HDF5 can freeze
+ * deal.II. If one MPI process throws an exception and not the other ones then
+ * deal.II freezes. The reason is that the following happens:
+ * 1. On the processor that calls the exception we try to close the parallel
+ *   MPI file (H5F_try_close) which ultimately results in a barrier
+ *   (PMPI_barrier). This is triggered by the destructor of HDF5Object.
```

Comment:
`PMPI_barrier` -> `PMPI_Barrier` ?

Also, can you confirm that the "P" prefix is correct. I would have (naively) thought that this should be `MPI_Barrier`.